### PR TITLE
calling dataloader every epoch now

### PR DESCRIPTION
**Pull request by williamFalcon**
Calls each dataloader every epoch as needed. If using the data_loader decorator then data loader will be init once per session. Otherwise, it'll be init on every epoch

### CODE REVIEW

**Comment by akhti** [comment]
Maybe rename `data_loader` decorator to `create_once`?

**Comment by jeffling** [comment]
This change would be very useful for us - is there anything we could do to push this over the finish line?

**Comment by williamFalcon** [comment]
yeah! this unfortunately worked before the 0.5 refactor. Maybe apply the changes manually to a new fresh fork from master?

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -601,7 +598,29 @@ def get_dataloaders(self, model):
...
+
+    def init_val_dataloader(self, model):
+        """
+        Dataloaders are provided by the model
+        :param model:
+        :return:
+        """
+        self.val_dataloader = model.val_dataloader
+
+        # enable multiple validation loaders
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py]: all this looks very similar for both test/valid... consider have a general func:
```
@staticmethod
def _nb_batches(dataloader, percent_check):
     have_loaders = dataloader is not None
     if have_loaders and not isinstance(dataloader, list):
         dataloader = [self.dataloader]

    # determine number of validation batches, datasets could be none, 1 or 2+
    if dataloader is not None:
        nb_batches = sum(len(dl) for dl in dataloader)
        nb_batches = int(nb_batches * percent_check)
        nb_batches = max(1, nb_batches)
    else:
        nb_batches = 0
    return dataloader, nb_batches
```
and later just:
```
self.val_dataloader, self.nb_val_batches = self._nb_batches(self.val_dataloader, self.val_percent_check)
```
and alternatively for `test` so it is clear thet the process is the same...

File: tox.ini [hunk 2]
```diff
@@ -44,4 +44,4 @@ doctests = True
 verbose = 2
 # https://pep8.readthedocs.io/en/latest/intro.html#error-codes
 format = pylint
-max-line-length = 100
+max-line-length = 120
```
Comment (Borda) [inline tox.ini]: do NOT change tests/formating!

**Pull request closed without merging**